### PR TITLE
rename max_redirect_count to max_retry_count

### LINKED ISSUE #41: clarify redirect vs retry error messages and variables
Recently my Redis cluster was down (getting a `CLUSTERDOWN` reply), but hiredis-cluster showed the following error: `too many cluster redirect`.

Looking at the code, ie. https://github.com/Nordix/hiredis-cluster/blob/76b3d67a80db3e7c08c5aba32046d63383b3490f/hircluster.c#L4043 it's comparing `cad->retry_count` with `cc->max_redirect_count`, and setting the error as `REDIS_ERR_CLUSTER_TOO_MANY_REDIRECT`.

Retries can happen for other reasons as well (like `CLUSTERDOWN`), so I suggest to change `max_redirect_count` to `max_retry_count` and update the error messages (I assume initially there was only a retry on redirects?)

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
Resolves #41

I'm not 100% happy with the error message "too many cluster retries", do you have better suggestions? Maybe "Maximum number of retries reached", "Request unsuccessful: Maximum number of retries reached" or something?

### CODE REVIEW

**Review by zuiderkwast — COMMENTED** [review]
No strong opinion but "too many retries" is fine by me. Short and simple.

In one place you wrote "redirections". Even if the last retry was a redirect, there may have been other retries before, so I though we should just say "retries" everywhere. Not a big deal though.

File: hircluster.c [hunk 1]
```diff
@@ -2454,9 +2454,9 @@ static void *redis_cluster_command_execute(redisClusterContext *cc,
     error_type = cluster_reply_error_type(reply);
     if (error_type > CLUSTER_NOT_ERR && error_type < CLUSTER_ERR_SENTINEL) {
         cc->retry_count++;
-        if (cc->retry_count > cc->max_redirect_count) {
-            __redisClusterSetError(cc, REDIS_ERR_CLUSTER_TOO_MANY_REDIRECT,
-                                   "too many cluster redirect");
+        if (cc->retry_count > cc->max_retry_count) {
+            __redisClusterSetError(cc, REDIS_ERR_CLUSTER_TOO_MANY_RETRIES,
+                                   "too many cluster redirections");
```
[the remark this answers is not in the document]
Comment (andreasgerstmayr) [reply hircluster.c:2459]: you're right, could have been other errors before - I'll change it

**Review by zuiderkwast — APPROVED** [review]
LGTM

**Review by bjosv — APPROVED** [review]
LGTM, thanks!

**Pull request merged**